php-cs-fixer

```diff
@@ -11,28 +13,33 @@
  * with this source code in the file LICENSE.
  */
 
+use Composer\XdebugHandler\XdebugHandler;
+use PhpCsFixer\Console\Application;
+
 error_reporting(E_ALL & ~E_DEPRECATED & ~E_USER_DEPRECATED);
 
-set_error_handler(static function ($severity, $message, $file, $line) {
-    if ($severity & error_reporting()) {
+set_error_handler(static function (int $severity, string $message, string $file, int $line): bool {
+    if (0 !== ($severity & error_reporting())) {
         throw new ErrorException($message, 0, $severity, $file, $line);
     }
+
+    return true;
 });
 
 // check environment requirements
-(function () {
-    if (\PHP_VERSION_ID === 80000) {
+(static function (): void {
+    if (\PHP_VERSION_ID === (int) '80000') {
         fwrite(STDERR, "PHP CS Fixer is not able run on PHP 8.0.0 due to bug in PHP tokenizer (https://bugs.php.net/bug.php?id=80462).\n");
         fwrite(STDERR, "Update PHP version to unblock execution.\n");
 
         exit(1);
     }
 
-    if (\PHP_VERSION_ID < 70400 || \PHP_VERSION_ID >= 80400) {
+    if (\PHP_VERSION_ID < (int) '70400' || \PHP_VERSION_ID >= (int) '80400') {
         fwrite(STDERR, "PHP needs to be a minimum version of PHP 7.4.0 and maximum version of PHP 8.3.*.\n");
         fwrite(STDERR, 'Current PHP version: '.PHP_VERSION.".\n");
 
-        if (getenv('PHP_CS_FIXER_IGNORE_ENV')) {
+        if (filter_var(getenv('PHP_CS_FIXER_IGNORE_ENV'), FILTER_VALIDATE_BOOLEAN)) {
```

Comment:
strictly speaking BC break, eg for `PHP_CS_FIXER_IGNORE_ENV=ignore_env` value.
i like the change, but i see it as candidate for next major, and now TODO or deprecation only

Reply:
@Wirone you are the author of this change, please adjust

Reply:
@keradus [official docs](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/c808dec27628cf82afff3af7bbdd4fceb2712eae/doc/usage.rst#environment-options) shows the usage as `PHP_CS_FIXER_IGNORE_ENV=1` so in my opinion we're safe to change this.

Reply:
unfortunately it's example and not spec/typehint. we should provide type info. we didn't? it's mistake. probably mine. still, we should not break it.

at least have a error msg when provided value is unexpected (optionally with TODO to remove in next major)

Reply:
IMHO the flag clearly reads as booleanish, personally I would consider it as a bug that needs to be fixed to ensure proper usage. Also, just look at [usages](https://github.com/search?q=PHP_CS_FIXER_IGNORE_ENV&type=code), flag is always set with `1` or `true`, which will still work after this change.

Reply:
I agree because `PHP_CS_FIXER_IGNORE_ENV` use is rare anyway. Let's not complicate our lifes too much.